### PR TITLE
UI branch cleanup

### PR DESCRIPTION
**Pull request by nostrbuddha**
- Android full screen issue (https://github.com/bisq-network/bisq-mobile/issues/61) - Made the system bar transparent, so it takes up current screen's BG color

 - Improved docs (README.md) for /components and /navigation

 - Component improvements: Button, TextField, Icons, PagerView, Layout components

 - Code cleanup: Create profile, Onboarding, TrustedNodeSetup
 
 - Clean, separate RootNavGraph and TabNavGraph

 - TopBar for Tab pages - Moved to TabContainerHome (added comments on why?)

 - Different onboarding page content in androidNode and xClient (facing some other issue in androidNode)

### CODE REVIEW

**Comment by rodvar** [comment]
rebased against main

File: bisqapps/androidNode/build.gradle.kts [hunk 1]
```diff
@@ -122,6 +123,7 @@ protobuf {
 dependencies {
     implementation(project(":shared:presentation"))
     implementation(project(":shared:domain"))
+    implementation(libs.navigation.compose)
```
Comment (rodvar) [inline bisqapps/androidNode/build.gradle.kts]: this is not needed, at least not in this pr..?

File: bisqapps/androidNode/build.gradle.kts [hunk 2]
```diff
@@ -26,6 +26,7 @@ kotlin {
     sourceSets {
         androidMain.dependencies {
             implementation(compose.preview)
+            implementation(libs.navigation.compose)
```
Comment (rodvar) [inline bisqapps/androidNode/build.gradle.kts]: this is not needed, at least not in this pr..?

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/GettingStartedScreen.kt [hunk 3]
```diff
@@ -34,15 +35,17 @@ import bisqapps.shared.presentation.generated.resources.icon_tag_outlined
...
 import network.bisq.mobile.presentation.ui.components.atoms.BisqText
+import network.bisq.mobile.presentation.ui.components.layout.BisqScrollLayout
 import network.bisq.mobile.presentation.ui.theme.*
 import org.jetbrains.compose.resources.DrawableResource
 import org.jetbrains.compose.resources.ExperimentalResourceApi
 import org.jetbrains.compose.resources.painterResource
 import org.koin.compose.koinInject
 
-interface IGettingStarted {
+interface IGettingStarted: ViewPresenter {
```
Comment (rodvar) [inline bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/GettingStartedScreen.kt:48]: nice!

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/GettingStartedScreen.kt [hunk 4]
```diff
@@ -51,73 +54,71 @@ interface IGettingStarted {
...
     // TODO attach view should happen here to let the presenter know?
+    // buddha: So here we have 2 screens actually.
+    //  - TabContainerScreen that has Scaffold, Topbar, Bottom bar (with 4 tabs)
+    //  - Indidivual tab screens like current GettingStartedScreen, which is technically
+    //  - not an indpendent screen. Since it doesn't have scaffold, topbar on it's own.
+
+    //  - So I guess, TabContainerScreen will have it's own TabContainerPresenter
+    //  - GettingStartedScreen (and other 3 tabs) will have it's own IGettingStarted
+    LaunchedEffect(Unit) {
+        presenter.onViewAttached()
```
Comment (rodvar) [inline bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/GettingStartedScreen.kt:70]: :+1:

File: bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/startup/SplashPresenter.kt [hunk 5]
```diff
@@ -30,18 +30,11 @@ open class SplashPresenter(
     }
 
     private fun navigateToNextScreen() {
-        // TODO: Conditional nav
+        // TODO: Conditional nav - Will implement once we got persistant storage from nish to save flags
```
Comment (rodvar) [inline bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/startup/SplashPresenter.kt:33]: nish is having some issues with his computer, I might be doing jumping on it this week :+1:

**Review by rodvar — APPROVED** [review]
Excellent work Buddha.

I fixed the android lint issue with android node and remove some libs , probably added when we didn't have the rootNavigator avail? please let me know if that's ok.

**Comment by nostrbuddha** [comment]
> Excellent work Buddha.
> 
> I fixed the android lint issue with android node and remove some libs , probably added when we didn't have the rootNavigator avail? please let me know if that's ok.

Yes. That's right. Thanks

**Pull request merged**